prody/dynamics/plotting.py

```diff
@@ -2381,3 +2355,43 @@ def showTree_networkx(tree, node_size=20, node_color='red', node_shape='o',
         showFigure()
 
     return mpl.gca()
+
+
+def checkColors(colors, num, labels, allowNumbers=False):
+    """Check colors and process them if needed"""
+
+    from matplotlib.colors import is_color_like
+
+    colors_dict = {}
+
+    if isinstance(colors, np.ndarray):
+        colors = tuple(colors)
+
+    if is_color_like(colors) or colors is None:
+        colors = [colors] * num
+    elif isListLike(colors):
+        colors = list(colors)
+
+    if isinstance(colors, dict):
+        if labels is None:
+            raise TypeError('color must be a string or a list unless labels are provided')
+        colors_dict = colors
+        colors = [colors_dict[label] for label in labels]
+
+    if isinstance(colors, list):
+        if len(colors) != num and not is_color_like(colors):
+            raise ValueError('colors should have the length of the set to be colored or satisfy matplotlib color rules')
+
+        if np.any([not is_color_like(color) for color in colors]):
+            if not allowNumbers:
+                raise ValueError('each element of colors should satisfy matplotlib color rules')
+            elif np.any([not isinstance(color, Number) for color in colors]):
+                raise ValueError('each element of colors should be a number or satisfy matplotlib color rules')
+
+        if len(colors) > 1 and np.any([not isinstance(color, type(colors[0])) for color in colors]):
+            raise TypeError('each element of colors should have the same type')
+
+    elif not ((allowNumbers and isinstance(colors, Number)) or is_color_like(colors)):
```

Comment:
Do we need another `is_color_like(colors)` check here?

Reply:
Ok, I think we just need to add the Number into the check to put it in a list of length num and then it should be checked in the list check

Reply:
Probably we still want a check for what if it's not a list from all this processing?